libs/core/tests/sharing_tests.rs

```diff
@@ -329,6 +329,213 @@ fn write_document_in_rejected_shared_folder_in_rejected_share_folder() {
     let result = cores[1].write_document(document.id, b"document content by sharee");
     assert_matches!(result.unwrap_err().kind, CoreError::InsufficientPermission);
 }
+#[test]
+fn write_link_by_sharee() {
+    let cores: Vec<Core> = vec![test_core_with_account(), test_core_with_account()];
+    let accounts = cores
+        .iter()
+        .map(|core| core.get_account().unwrap())
+        .collect::<Vec<_>>();
+
+    let document1 = cores[0].create_at_path("/document1").unwrap();
+
+    cores[0]
+        .share_file(document1.id, &accounts[1].username, ShareMode::Write)
+        .unwrap();
+
+    cores[0].sync(None).unwrap();
+    cores[1].sync(None).unwrap();
+
+    let link = cores[1]
+        .create_link_at_path("/link1", document1.id)
+        .unwrap();
+    cores[1]
+        .write_document(link.id, b"document content by sharee")
+        .unwrap();
+
+    cores[1].sync(None).unwrap();
+    cores[0].sync(None).unwrap();
+
+    assert_eq!(cores[1].read_document(link.id).unwrap(), b"document content by sharee");
+    assert_eq!(cores[0].read_document(document1.id).unwrap(), b"document content by sharee");
```

Comment:
Getting the link back is the most reasonable behavior for `create_link_at_path`, but I believe you're not going to want to `read_document` and accept a `link`. I think trying to `read_document` and passing a link is probably an error. Ideally you would tweak this case to make sure that you can read the document by both people, and then I wonder if this test exists already.

Reply:
okay i removed link swapping on `read_document`

Reply:
yeah the test already exists on `edited_document` ``sync_service_share_tests`